src/Button.js

```diff
@@ -1,34 +1,82 @@
+import React from 'react';
 import styled, { css } from 'react-emotion';
 import PropTypes from 'prop-types';
 import { rem, transparentize } from 'polished';
 
-const sizeStyles = ({ theme, size }) => {
+const sizeStyles = ({ theme, size, text }) => {
   const sizeMap = {
     small: {
       padding: `0 ${theme.space.smaller}`,
       height: rem('24px'),
-      fontSize: theme.fontSizes.size1
+      fontSize: theme.fontSizes.size1,
+      i: {
+        fontSize: rem('24px'),
+        verticalAlign: text === undefined ? 0 : '-4px'
```

Comment:
Not a huge fan of these magic variables used for the alignment inside the button. I know sometimes getting things to align can be a bit of a crapshoot. If nothing else can we turn these into variables since both the -4px and -6px both appear to be used twice.

Reply:
Yeah, it was a bit tricky getting the vertical alignment in place here. Agreed on the magic numbers. I think my only issue was getting a useful name for a variable here. Do you have any suggestions? Maybe just `verticalAlignSmall` and `verticalAlignSmaller`?

Reply:
Those work for me, give at least some context, and prevent a rube like me from coming in an accidentally changing one and not the other.

Reply:
Yeah, constantizing any repeated values would help. Because it's not currently clear how many of those repetitions are coincidental vs. intentionally the  "same value".